### PR TITLE
Add details about some RPCs

### PR DESCRIPTION
**Pull request by teor2345**
I found some things that we need to do, and some things we don't. Here are my suggestions.

### CODE REVIEW

File: zebra/explorer.md [hunk 1]
```diff
@@ -13,23 +13,23 @@ https://zcashblockexplorer.com/
...
+| getblockhashes              | NO.                | As it is used in different places, with different arguments, it need all the fields and options.<br>(noOrphans is always true.)                                                                                                                    | [Zcash](https://zcash.github.io/rpc/getblockhashes.html)                                                                                                                     | [1](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer/blocks/block_warmer.ex#L19) - [2](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer/transactions/transaction_cache.ex#L19) - [3](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/controllers/block_controller.ex#L54) - [4](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/controllers/block_controller.ex#L89)                                                                                                                              |
```
Comment (teor2345) [inline zebra/explorer.md:16]: Added: (noOrphans is always true.)

Seems like the word diff isn't working on GitHub, probably due to the whitespace.

File: zebra/explorer.md [hunk 2]
```diff
@@ -13,23 +13,23 @@ https://zcashblockexplorer.com/
...
...e=2` mode:<br>`height`<br>`size`<br>`hash`<br>`time`<br>`tx.vout`<br>We'll need to add `time`, `size`, and `tx.vout` to the existing database indexes, and `size` to the non-finalized state indexes.<br>                                                                                                  | [Zcash](https://zcash.github.io/rpc/getblock.html) - [Zebra](https://doc.zebra.zfnd.org/zebra_rpc/methods/trait.Rpc.html#tymethod.get_block)                                 | [1](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer/blocks/block_warmer.ex#L24) [calls](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/views/block_view.ex#L72) - [2](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer/transactions/transaction_cache.ex#L24) - [3](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/controllers/block_controller.ex#L5) - [4](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/controllers/block_controller.ex#L12) - [5](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/controllers/search_controller.ex#L14) |
```
Comment (teor2345) [inline zebra/explorer.md:17]: Changed to the RPC fields that are used by the block explorer, it calculates some of the final info itself. Added verbose mode and database info.

File: zebra/explorer.md [hunk 3]
```diff
@@ -13,23 +13,23 @@ https://zcashblockexplorer.com/
...
...upport this call, however the explorer use the verbose mode that zebra do not support yet.<br>We'd need to track when transactions entered the mempool:<br>`height`<br>`time`                                                                                                                     | [Zcash](https://zcash.github.io/rpc/getrawmempool.html) - [Zebra](https://doc.zebra.zfnd.org/zebra_rpc/methods/trait.Rpc.html#tymethod.get_raw_mempool)                      | [1](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer/mempool/mempool_warmer.ex#L14) [called by](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/live/raw_mempool_live.ex#L19)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |
```
Comment (teor2345) [inline zebra/explorer.md:18]: Added extra mempool code that would be needed.

File: zebra/explorer.md [hunk 4]
```diff
@@ -13,23 +13,23 @@ https://zcashblockexplorer.com/
...
+| getinfo                     | YES.    | Zebra fully implement the arguments and results of this call. | [Zcash](https://zcash.github.io/rpc/getinfo.html) - [Zebra](https://doc.zebra.zfnd.org/zebra_rpc/methods/trait.Rpc.html#tymethod.get_info)                                   | [1](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer/metrics/info_warmer.ex#L14)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
```
Comment (teor2345) [inline zebra/explorer.md:19]: Moved fields to getblockchaininfo

File: zebra/explorer.md [hunk 5]
```diff
@@ -13,23 +13,23 @@ https://zcashblockexplorer.com/
...
+| getmempoolinfo              | NO.                | No arguments.<br>We'll need to estimate mempool memory usage.                                                                                                                                                                                                       | [Zcash](https://zcash.github.io/rpc/getmempoolinfo.html)                                                                                                                     | [1](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer/metrics/mempool_info_warmer.ex#L14)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |
```
Comment (teor2345) [inline zebra/explorer.md:20]: Added the mempool memory usage estimate for that field.

File: zebra/explorer.md [hunk 6]
```diff
@@ -13,23 +13,23 @@ https://zcashblockexplorer.com/
...
...d:<br>`commitments`<br>`difficulty`<br>`size_on_disk`<br>`valuePools`<br><br>These fields are missing in the zebra implementation.<br>We'll need to add a new database entry for the total block size.<br>In the explorer this information is appended to an `info` array.                                                                                                                                                    | [Zcash](https://zcash.github.io/rpc/getblockchaininfo.html) - [Zebra](https://doc.zebra.zfnd.org/zebra_rpc/methods/trait.Rpc.html#tymethod.get_blockchain_info)              | [1](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer/metrics/metrics_warmer.ex#L14)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |
```
Comment (teor2345) [inline zebra/explorer.md:21]: Move fields from getinfo, deleted fields we already implement.

File: zebra/explorer.md [hunk 7]
```diff
@@ -13,23 +13,23 @@ https://zcashblockexplorer.com/
...
+| getpeerinfo                 | YES, partially.    | The explorer takes all the fields and add them to a `nodes`  array. The following extra fields are needed in Zebra:<br>`subver`<br>`synced_blocks`                                       | [Zcash](https://zcash.github.io/rpc/getpeerinfo.html) - [Zebra](https://doc.zebra.zfnd.org/zebra_rpc/methods/trait.GetBlockTemplateRpc.html#tymethod.get_peer_info)          | [1](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer/nodes/nodes_warmer.ex#L14) [called by](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/live/nodes_live.ex#L17)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
```
Comment (teor2345) [inline zebra/explorer.md:23]: Added the 2 new fields we need.

File: zebra/explorer.md [hunk 8]
```diff
@@ -13,23 +13,23 @@ https://zcashblockexplorer.com/
...
...d but at least the following extra fields are needed in the response:<br>`txid`<br>`time` (from block header)<br>`size`<br>`vout`<br>tx type: many transparent and shielded fields<br>We'll need to add `tx.size` to the existing database and non-finalized state indexes.               | [Zcash](https://zcash.github.io/rpc/getrawtransaction.html) - [Zebra](https://doc.zebra.zfnd.org/zebra_rpc/methods/trait.Rpc.html#tymethod.get_raw_transaction)              | [1](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer/transactions/transaction_cache.ex#L30) - [2](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/controllers/search_controller.ex#L15) - [3](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/controllers/transaction_controller.ex#L5) [called by](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/live/recent_transactions_live.ex#L19) - [4](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/controllers/transaction_controller.ex#L12)                                                                                                     |
```
Comment (teor2345) [inline zebra/explorer.md:24]: Changed to the RPC fields used by the block explorer, removed the ones it calculates itself. Unfortunately we need a lot of fields for the explorer to do the transaction type.

File: zebra/explorer.md [hunk 9]
```diff
@@ -13,23 +13,23 @@ https://zcashblockexplorer.com/
...
...ra needs to implement the `received` field.<br>This needs to be stored for each address in the database, as a `u128`.                                                                                  | [Zcash](https://zcash.github.io/rpc/getaddressbalance.html) - [Zebra](https://doc.zebra.zfnd.org/zebra_rpc/methods/trait.Rpc.html#tymethod.get_address_balance)              | [1](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/controllers/address_controller.ex#L28) - [2](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/controllers/address_controller.ex#L72) [calls](https://github.com/nighthawk-apps/zcashex/blob/main/lib/zcashex.ex#L128) [called by](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/templates/address/address.html.eex#L45)                                                                                                                                                                                                                                                                                                                                                                     |
```
Comment (teor2345) [inline zebra/explorer.md:25]: The arguments are always an array, `zcashd` doesn't accept a single string. (The docs are wrong!) Linked to the code that changes a single argument to an array.

We also need an extra field here.

File: zebra/explorer.md [hunk 10]
```diff
@@ -13,23 +13,23 @@ https://zcashblockexplorer.com/
...
+| getaddressdeltas            | NO.                | We need to implement almost fully.<br>This might need a new index in the database.                                                                                                                                                                                  | [Zcash](https://zcash.github.io/rpc/getaddressdeltas.html)                                                                                                                   | [1](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/controllers/address_controller.ex#L29) - [2](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/controllers/address_controller.ex#L73)                                                                                                                                                                                                                                                                                                                                                                     |
```
Comment (teor2345) [inline zebra/explorer.md:26]: Added database info.

File: zebra/explorer.md [hunk 11]
```diff
@@ -13,23 +13,23 @@ https://zcashblockexplorer.com/
...
+| z_listunifiedreceivers      | MINING.                | Not implemented but I'll implement as part of the mining project: https://github.com/ZcashFoundation/zebra/issues/6030                                                                                         | [Zcash](https://zcash.github.io/rpc/z_listunifiedreceivers.html)                                                                                                             | [1](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/controllers/address_controller.ex#L96)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
```
Comment (teor2345) [inline zebra/explorer.md:27]: Changed to "MINING" so we know it will be done before we start the explorer work.

File: zebra/explorer.md [hunk 12]
```diff
@@ -13,23 +13,23 @@ https://zcashblockexplorer.com/
...
+| z_validatepaymentdisclosure | NO.                | Need to be fully implemented. Specification still in draft.                                                                                                                           | [Zcash](https://zcash.github.io/rpc/z_validatepaymentdisclosure.html) [docs](https://github.com/zcash/zcash/blob/master/doc/payment-disclosure.md) [draft ZIP](https://github.com/zcash/zips/pull/119)                                                                                                       | [1](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/controllers/page_controller.ex#L49)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
```
Comment (teor2345) [inline zebra/explorer.md:30]: This is a draft ZIP, maybe we could just skip this RPC? It hasn't been updated for Orchard yet, and it needs some format changes to work with NU5.

File: zebra/explorer.md [hunk 13]
```diff
@@ -13,23 +13,23 @@ https://zcashblockexplorer.com/
...
+| z_validateaddress           | MINING.                | Will be implemented as part of the mining work: https://github.com/ZcashFoundation/zebra/issues/6083                                                                                                                                                                                           | [Zcash](https://zcash.github.io/rpc/z_validateaddress.html)                                                                                                                  | [1](https://github.com/nighthawk-apps/zcash-explorer/blob/main/lib/zcash_explorer_web/controllers/search_controller.ex#L17)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
```
Comment (teor2345) [inline zebra/explorer.md:32]: These are both mining RPCs, no need to worry about them!

**Review by teor2345 — COMMENTED** [review]
Looks great, I added some details, but it's a good initial plan.

Do you know if the Zcash explorer website fails totally when RPCs are missing? Or does only the page or specific text fail?

If we can do one explorer page at a time, it will be a lot easier. We can decide that some pages aren't a priority, and do them later.

**Review by oxarbitrage — APPROVED** [review]
Thank you very much for the changes, they all add value.

**Pull request merged**